src/main/java/reposense/wizard/InputBuilder.java

```diff
@@ -0,0 +1,289 @@
+        input.append(ArgsParser.CLONING_THREADS_FLAG[0] + WHITESPACE + threads + WHITESPACE);
+        return this;
+    }
+
+    /**
+     * Adds the analysis threads flag with the {@code threads} as argument to the input.
+     * This method should only be called once in one build.
+     *
+     * @param threads The number of threads for analysis.
+     */
+    public InputBuilder addNumAnalysisThreads(int threads) {
+        input.append(ArgsParser.ANALYSIS_THREADS_FLAG[0] + WHITESPACE + threads + WHITESPACE);
+        return this;
+    }
+
+    /**
+     * Adds the flag to enable shallow cloning.
+     * This method should only be called once in one build.
+     */
+    public InputBuilder addShallowCloning() {
+        input.append(ArgsParser.SHALLOW_CLONING_FLAGS[0] + WHITESPACE);
+        shallowCloning = true;
+        return this;
+    }
+
+    /**
+     * Adds the flag to enable find previous authors.
+     * This method should only be called once in one build.
+     */
+    public InputBuilder addFindPreviousAuthors() {
+        input.append(ArgsParser.FIND_PREVIOUS_AUTHORS_FLAGS[0] + WHITESPACE);
+        return this;
+    }
+
+    /**
+     * Adds the flag to enable test mode.
+     * This method should only be called once in one build.
+     */
+    public InputBuilder addTestMode() {
+        input.append(ArgsParser.TEST_MODE_FLAG[0] + WHITESPACE);
+        return this;
+    }
+
+    /**
+     * Adds the flag to include modified date in lines.
+     * This method should only be called once in one build.
+     */
+    public InputBuilder addLastModifiedDateFlags() {
+        input.append(ArgsParser.LAST_MODIFIED_DATE_FLAGS[0] + WHITESPACE);
+        return this;
+    }
+
+    /**
+     * Adds the flag to enable fresh cloning.
+     * This method should only be called once in one build.
+     */
+    public InputBuilder addFreshCloning() {
+        input.append(ArgsParser.FRESH_CLONING_FLAG[0] + WHITESPACE);
+        return this;
+    }
+
+    /**
+     * Adds {@code content} to the input.
+     */
+    public InputBuilder add(String content) {
+        input.append(content + WHITESPACE);
+        return this;
+    }
+
+    /**
+     * Adds {@code num} of white spaces to the input.
+     *
+     * @param num The number of white spaces to add.
+     */
+    public InputBuilder addWhiteSpace(int num) {
+        for (int i = 0; i < num; i++) {
+            input.append(WHITESPACE);
+        }
+        return this;
+    }
+
+    /**
+     * Clears all input and flags given.
+     */
+    public InputBuilder reset() {
+        init();
+        return this;
+    }
+
+    private static String addQuotationMarksToPath(String path) {
+        return '"' + path + '"';
+    }
+
+    private static String addQuotationMarksToPath(Path path) {
+        return addQuotationMarksToPath(path.toString());
+    }
+
+    public boolean isShallowCloning() {
+        return shallowCloning;
+    }
```

Comment:
I think we should only add the ones we are currently using into this class and delete the remaining. I don't think this feature will grow to implement many of the other flags

Reply:
Got it, we will clean up the unused `InputBuilder.java` functions.